sections/futures/UserInfo/PositionsTable.tsx

```diff
@@ -0,0 +1,409 @@
+import LinkArrow from 'assets/svg/app/link-arrow.svg';
+import Button from 'components/Button';
+import Currency from 'components/Currency';
+import { FlexDivRowCentered } from 'components/layout/flex';
+import { DesktopOnlyView, MobileOrTabletView } from 'components/Media';
+import Pill from 'components/Pill';
+import Table, { TableNoResults } from 'components/Table';
+import { Body, NumericValue } from 'components/Text';
+import { EXTERNAL_LINKS } from 'constants/links';
+import { NO_VALUE } from 'constants/placeholder';
+import ROUTES from 'constants/routes';
+import useIsL2 from 'hooks/useIsL2';
+import useNetworkSwitcher from 'hooks/useNetworkSwitcher';
+import { FuturesAccountType } from 'queries/futures/subgraph';
+import MobilePositionRow from 'sections/dashboard/FuturesPositionsTable/MobilePositionRow';
+import PositionType from 'sections/futures/PositionType';
+import {
+	selectCrossMarginPositions,
+	selectIsolatedMarginPositions,
+	selectMarketAsset,
+	selectMarkets,
+	selectPositionHistory,
+} from 'state/futures/selectors';
+import { useAppSelector } from 'state/hooks';
+import media from 'styles/media';
+
+type FuturesPositionTableProps = {
+	accountType: FuturesAccountType;
+	showCurrentMarket?: boolean;
+	showEmptyTable?: boolean;
+};
+
+const LegacyLink = () => {
+	const { t } = useTranslation();
+	const theme = useTheme();
+	return (
+		<ButtonContainer>
+			<Button
+				fullWidth
+				variant="flat"
+				size="small"
+				noOutline={true}
+				textTransform="none"
+				onClick={() => window.open(EXTERNAL_LINKS.Trade.V1, '_blank', 'noopener noreferrer')}
+			>
+				{t('dashboard.overview.futures-positions-table.legacy-link')}
+				<StyledArrow fill={theme.colors.selectedTheme.text.value} />
+			</Button>
+		</ButtonContainer>
+	);
+};
+
+const PositionsTable: FC<FuturesPositionTableProps> = ({
+	accountType,
+	showCurrentMarket = true,
+	showEmptyTable = true,
+}) => {
+	const { t } = useTranslation();
+	const router = useRouter();
+	const { switchToL2 } = useNetworkSwitcher();
+
+	const isL2 = useIsL2();
+
+	const isolatedPositions = useAppSelector(selectIsolatedMarginPositions);
+	const crossMarginPositions = useAppSelector(selectCrossMarginPositions);
+	const positionHistory = useAppSelector(selectPositionHistory);
+	const currentMarket = useAppSelector(selectMarketAsset);
+	const futuresMarkets = useAppSelector(selectMarkets);
+
+	let data = useMemo(() => {
+		const positions = accountType === 'cross_margin' ? crossMarginPositions : isolatedPositions;
+		return positions
+			.map((position) => {
+				const market = futuresMarkets.find((market) => market.asset === position.asset);
+				const thisPositionHistory = positionHistory.find((ph) => {
+					return ph.isOpen && ph.asset === position.asset;
+				});
+
+				return {
+					market: market!,
+					position: position.position!,
+					avgEntryPrice: thisPositionHistory?.avgEntryPrice,
+					stopLoss: accountType === 'cross_margin' ? (position as any).stopLoss : undefined,
```

Comment:
Would be good if we could avoid `as any` here just so if we change something this wouldn't go unnoticed

Reply:
The issue here is that TS does not pick up the stopLoss and takeProfit values from the cross margin positions for some reason. I'll just add those to the isolated positions type as well then.